.pylintrc

```diff
@@ -1,17 +1,32 @@
 # https://docs.pylint.org/en/latest/technical_reference/features.html
 
+[MASTER]
+ignore=_version.py conf.py
+jobs=0
+
 [MESSAGES CONTROL]
-disable =
-  # erroneously flags pytest decorators like parametrize and skipif:
+disable=
+  abstract-method,
+  arguments-differ,
+  attribute-defined-outside-init,
+  bad-continuation,
+  broad-except,
+  invalid-name,
+  isinstance-second-argument-not-valid-type,  # https://github.com/PyCQA/pylint/issues/3507
+  line-too-long,
+  missing-function-docstring,  # prevents idiomatic type hints
+  multiple-statements,  # prevents idiomatic type hints
+  no-init,
+  no-member,
+  no-self-use,  # prevents idiomatic type hints
   not-callable,
-
-# bidict/_version.py is generated by setuptools_scm
-ignore = _version.py
-
-# Maximum number of parents for a class.
-# The default of 7 results in "too-many-ancestors" for all bidict classes.
-max-parents = 13
-
-# Maximum number of arguments for a function.
-# The default of 5 only leaves room for 4 args besides self for methods.
-max-args=6
+  protected-access,
```

Comment:
Does this need to be in this PR? Seems unrelated a lot of them.

Reply:
Thanks for taking a look at this PR!

Adding type hints to an entire codebase touches the entire codebase. In the process, I had to tune linter settings so I could spend less time fighting linters and be more productive. Specifically, linters fight with type hints and mypy in a couple ways: they lack good support composing line-based controls (e.g. `# noqa: E123; type: ignore; pylint: disable=foo`), and they flag idiomatic type hint style (e.g. for overloads).

In general, since I have had no one who is consistently available to review my changes to bidict (I am often the only one reviewing my PRs), and the code is all written by me alone on a voluntary basis on nights and weekends, it just hasn't been at all practical to optimize my PRs for the type of reviewer who prefers PRs to be split. I don't think I'm going to break this PR up into multiple PRs at this time. But it's helpful to know you prefer this in case you end up reviewing more of my PRs in the future.